From f9a15133eb065fe5711e861f7183e3691b3e15df Mon Sep 17 00:00:00 2001
From: Yuval Caduri <cyuval@marvell.com>
Date: Tue, 18 Jul 2017 10:07:08 +0300
Subject: [PATCH 1800/2241] fix: mvpp2x: net: Enable link-event for musdk port

This fix enables link-event interrupts for ports owned by musdk.
Required to indicate the correct link-state status for these ports.

Change-Id: I7fcee8ef72b611a2cf8cdba4ec3d9dae22bf96ce
Signed-off-by: Yuval Caduri <cyuval@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/41748
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index d42c251..812f8b8 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -3965,8 +3965,7 @@ int mv_pp2x_open(struct net_device *dev)
 		netdev_err(port->dev, "cannot allocate Tx queues\n");
 		goto err_cleanup_rxqs;
 	}
-	if (!(port->flags & MVPP2_F_IF_MUSDK))
-		err = mv_pp2x_setup_irqs(dev, port);
+	err = mv_pp2x_setup_irqs(dev, port);
 	if (err) {
 		netdev_err(port->dev, "cannot allocate irq's\n");
 		goto err_cleanup_txqs;
@@ -5212,8 +5211,7 @@ skip_tso_buffers:
 	mv_gop110_mib_counters_clear(&port->priv->hw.gop, port->mac_data.gop_index);
 	mv_pp2x_counters_stat_clear(port);
 
-	if (!(port->flags & MVPP2_F_IF_MUSDK))
-		mv_pp2x_port_irq_names_update(port);
+	mv_pp2x_port_irq_names_update(port);
 
 	if (priv->pp2_version == PPV22)
 		port->port_hotplug_nb.notifier_call = mv_pp2x_port_cpu_callback;
-- 
2.7.4

